main/observing/observation_run.py

```diff
@@ -630,3 +673,58 @@ def _shutdown_procedure(self, calibration, cooler=True):
             self.take_calibration_images()
         if cooler:
             self.camera.onThread(self.camera.cooler_set, False)
+
+    def threadcheck(self):
+        '''
+        Description
+        ----------
+        Checks to see if self.monitor has raised a crashed thread,
+        Restarts the crashed threads if there are any
+        
+        Returns
+        -------
+        None
+        '''
+        #print('---In thread Check')
+        threadlist = self.monitor.crash_return()
+        if self.monitor.threadcrash == True:
+            #print('Threads have crashed')
+            threadlist = self.monitor.crash_return()
+            for thname in threadlist:
+                self.threadrestart(thname)
+        else:
+            logging.info('All threads OK')
+
+    def restart(self, thname):
+        '''
+        Description
+        -----------
+        Redefines and then restarts the inputted thread
+        
+        Parmeters
+        --------
+        thname : Handle
+            Handle of the original thread to restart
+        '''
+
+        for x in self.th_dict:
+            print('In thread restart')
```

Comment:
why don't you just check if thname == each of these strings?  the whole for loop here is pointless

Reply:
better yet, you could just do self.th_dict[thname] = Camera() if thname == 'camera' else Telescope() if thname == 'telescope' else ....
and then just self.th_dict[thname].start()